### ISSUE TITLE
A false positive about the rule DM_GC

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#2033]
Hi, I found a false positive about the rule [DM_GC](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dm-explicit-garbage-collection-extremely-dubious-except-in-benchmarking-code-dm-gc). Please read the following two cases.

No warnings with this case and it is right.
```java
public static final boolean tag = false;
public void foo() {
  if(tag) {
    System.gc();
  }
}
```
However, the following case has a warning in line and these two cases are equivalent.
```java
public static final boolean tag;
static {
  tag = false;
}
public void foo() {
  if(tag) {
    System.gc();  // has a warning
  }
}
```
Hence, I think this is a false positive. Thanks for your consideration.

SpotBugs version: Latest
